### PR TITLE
numbers: include the Number Tower

### PR DESCRIPTION
**Pull request by fosskers**
As discussed on Discord. This adds an image to the top of the page that allows the reader to see the relationships between all the main number types at a glance.

### CODE REVIEW

File: numbers.md [hunk 1]
```diff
@@ -11,6 +11,14 @@ Some sources:
 * [`Numbers, Characters and Strings`][numbers-characters-strings]
   in Practical Common Lisp
 
+<!-- epub-exclude-start -->
```
Comment (fosskers) [inline numbers.md:14]: I borrowed this pragma from the index page.

File: numbers.md [hunk 2]
```diff
@@ -11,6 +11,14 @@ Some sources:
...
+<div style="text-align: center">
+    <img src="numbertower.png" alt="Number Types in Common Lisp"/>
+</div>
+<!-- epub-exclude-end -->
+
+*Types in bold, cyan boxes are the ones you will typically use.*
```
Comment (fosskers) [inline numbers.md:20]: Is this enough elaboration?

**Comment by fosskers** [comment]
Hi, any thoughts?

**Comment by kilianmh** [comment]
LGTM. It's a good starting point. Is ECL the only implementation with differing specific number type, or are there also others?

**Comment by fosskers** [comment]
Originally the image was much more complicated because it included all the weird hidden subtypes of various things (mainly children of `complex`) that you aren't actually expected to use yourself. I took them back out to make it clearer.

With regards to `long-float`, the precision of floats is implementation dependent, according to the spec: https://cl-community-spec.github.io/pages/short_002dfloat.html

As far as I found, `double-float` and `long-float` are usually the same. ECL in particular makes it extra long, based on C's `long double`.

File: numbers.md (hunk 2 above)
Comment (fosskers) [reply numbers.md:20]: I added more about ECL in the Floating Point section.

**Comment by fosskers** [comment]
Thanks.

**Comment by kilianmh** [comment]
We thank you for your efforts!

**Pull request merged**